gpu/src/Buffer.zig

```diff
@@ -103,9 +103,9 @@ pub const BindingType = enum(u32) {
 
 pub const BindingLayout = extern struct {
     reserved: ?*anyopaque = null,
-    type: BindingType,
-    has_dynamic_offset: bool,
-    min_binding_size: u64,
+    type: BindingType = .none,
```

Comment:
I think we should have this default to `.uniform` since that's what it's defined as in the spec:

<img width="564" alt="image" src="https://user-images.githubusercontent.com/3173176/162637495-99c752d2-ad6b-4d32-84d9-d562104dc062.png">

https://www.w3.org/TR/webgpu/#dictdef-gpubufferbindinglayout